src/steamship_langchain/file_loaders/sphinx.py

```diff
@@ -0,0 +1,114 @@
+import logging
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+from steamship import Block, File, MimeTypes, Steamship, Tag
+from steamship.data import TagKind, TagValueKey
+from steamship.data.tags.tag_constants import ProvenanceTag
+
+from steamship_langchain.file_loaders import BaseFileLoader
+
+
+def generate_url_tags(client: Steamship, url: str, metadata: Optional[Dict[str, str]] = None):
+    tags = [
+        Tag(
+            client=client,
+            kind=TagKind.TIMESTAMP,
+            name="timestamp",
+            value={"timestamp": datetime.now().isoformat()},
+        ),
+        Tag(
+            client=client,
+            kind=TagKind.PROVENANCE,
+            name=ProvenanceTag.URL,
+            value={TagValueKey.STRING_VALUE: url},
+        ),
+    ]
+    if metadata:
+        tags.append(Tag(client=client, kind="metadata", name="metadata", value=metadata))
+    return tags
+
+
+class SphinxSiteLoaderBase(BaseFileLoader):
+    """Load sphinx websites into Steamship.
+
+    Note:
+        Requires install of BeautifulSoup, via:
+        `pip install bs4`
+    """
+
+    tag_name: str
+    "Name of html tag to use for imports"
+
+    tag_attributes: Dict[str, str] = {}
+    "Identifying attributes for the tag"
+
+    scheme: str = "https://"
+
+    use_section_id_in_provenance: bool = True
+
+    def load(self, path: str, metadata: Optional[Dict[str, str]] = None) -> List[File]:
+        """Load documents."""
+        from bs4 import BeautifulSoup
+
+        def _clean_data(data: str) -> List[Tuple[str, str]]:
+            soup = BeautifulSoup(data, "html.parser")
+            doc_tags = soup.find_all(self.tag_name, self.tag_attributes)
+
+            if len(doc_tags) == 0:
+                return []
+
+            return [
+                (t.get_text(separator=" ", strip=True), t.get("id", "")) for t in doc_tags if t.text
+            ]
+
+        files = []
+        for p in Path(path).rglob("*.html"):
+            if p.is_dir():
+                continue
+            with open(p) as f:
```

Comment:
```suggestion
            with p.open() as f:
```

Not sure if open(Path) is recommended I thought this would crash.

Reply:
Didn't crash, but i've changed per recommendation.